cluster/gce/gci/configure-helper.sh

```diff
@@ -182,13 +182,14 @@ function get-or-generate-uuid(){
   fi
 
   # each line of the ssdmap looks like "${device} persistent-uuid"
-  if [[ ! -z $(grep ${device} ${ssdmap}) ]]; then
+  if grep -q "${device}" "${ssdmap}"; then
     #create symlink based on saved uuid
-    local myuuid=$(grep ${device} ${ssdmap} | cut -d ' ' -f 2)
+    local myuuid
+    myuuid=$(grep "${device}" "${ssdmap}" | cut -d ' ' -f 2)
   else
     # generate new uuid and add it to the map
-    local myuuid=$(uuidgen)
-    if [[ ! ${?} -eq 0 ]]; then
+    local myuuid
+    if ! myuuid=$(uuidgen); then
```

Comment:
Does it make sense to simplify this slightly by moving the variable declaration outside the `if` statement, and then only worry about assignment within the if/else?

Reply:
I'm all for simplification, but I don't quite follow what you mean. Could you please explain with code to make it absolutely clear?

Reply:
```
# each line of the ssdmap looks like "${device} persistent-uuid"
local myuuid
if grep -q "${device}" "${ssdmap}"; then
     #create symlink based on saved uuid
     myuuid=$(grep "${device}" "${ssdmap}" | cut -d ' ' -f 2)
else
     # generate new uuid and add it to the map
     if ! myuuid=$(uuidgen); then
         echo "Failed to generate valid UUID with uuidgen" >&2
         exit 2
     fi
....
```

Reply:
Ah, yes, now I see. Yes, of course, very good suggestion. I will absolutely fix this.